numpy/distutils/ccompiler.py

```diff
@@ -191,7 +230,8 @@ def CCompiler_compile(self, sources, output_dir=None, macros=None,
 
     def single_compile(args):
         obj, (src, ext) = args
-        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
+        if _needs_build(obj):
+            self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
```

Comment:
Deleting might not help, because the same file is built two times with different options during the build.